online-shop/src/app/cart/cart.component.ts

```diff
@@ -13,25 +19,34 @@ import { trigger } from '@angular/animations';
 })
 export class CartComponent implements OnInit {
 
-  constructor(private cartService : CartService, private productService: ProductService) { }
-  cart: IProductWrapper[];
-  products: IProduct[] = [];
+  constructor(private store: Store<AppState>, private storeProduct: Store<AppState>, private storeOrder: Store<AppState>) { }
+  cart: IProductWrapper[] = [];
+  products: IProduct[]  = [];
   observableProducts: Observable<IProduct>[] = [];
   productSubscription : Subscription;
+  public product$ = this.store.select(selectOneProduct);
+  addedProduct: IProduct | undefined;
   ngOnInit(): void {
-      this.cart = this.cartService.getCart();
       this.allProductsFromChart();
   }
 
   allProductsFromChart(){
+    this.products = []
+    this.store.select(selectAllCart).subscribe((response) => {
+    this.cart = response;
+    });
+
     this.cart.forEach(element => {
-            this.observableProducts.push(this.productService.getProductById(element.productId.toString()));
-          });
-    forkJoin(this.observableProducts).subscribe(data => this.products = data);
+      this.storeProduct.select(selectProductByID(element.productId)).pipe(take(1)).subscribe((product) => {
+        if(product !== undefined)  
+           this.products.push(product);
+    });
+  }) 
 }
-
   placeOrder(){
-   this.productSubscription = this.cartService.checkout().subscribe(() => alert("Order created successfully!"));
+ 
+   this.storeOrder.dispatch(checkout())
+    alert("Order created successfully!");
```

Comment:
This `alert()` can be quite intrusive for the end user. You can use either a [Dialog](https://material.angular.io/components/dialog/examples) or a [SnackBar](https://material.angular.io/components/snack-bar/overview) as a popup notification system, but it's not mandatory.